packages/runtime-dom/src/helpers/useCssVars.ts

```diff
@@ -99,8 +99,12 @@ function setVarsOnNode(el: Node, vars: Record<string, string>) {
     const style = (el as HTMLElement).style
     let cssText = ''
     for (const key in vars) {
-      style.setProperty(`--${key}`, vars[key])
-      cssText += `--${key}: ${vars[key]};`
+      const value = vars[key]
+      if (isRenderableAttrValue(value)) {
```

Comment:
A potential breaking change might have been introduced here. Previously, the client would render `undefined` value, effectively setting the rule value to `unset`. However, now `undefined` value will not be rendered, causing the rule to inherit value from the parent node. However, this is an edge case.
Notably, this change aligns with SSR behavior.

Reply:
Note: I don’t think rendering `undefined` was proper to render regardless; it was essentially setting the value to a string of `’undefined’` which simply doesn’t map to a value. 

This worked because undefined is never a valid value in CSS.
 
In these scenarios where we want to unset and prevent inheritance, it should actually be set to a value of `initial` instead, which unsets the value in the current scope. 

I was thinking you could use [`@property` syntax](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) to declare the custom property and control inheritance; however, this cannot be defined inline on an element.